src/axom/quest/tests/quest_sampling_shaper.cpp

```diff
@@ -0,0 +1,868 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \brief Unit tests for quest's SamplingShaper class and associated replacement rules.
+ */
+
+#include "gtest/gtest.h"
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/klee.hpp"
+#include "axom/primal.hpp"
+#include "axom/quest.hpp"
+#include "axom/sidre.hpp"
+#include "axom/slic.hpp"
+#include "axom/quest/SamplingShaper.hpp"
+
+#ifndef AXOM_USE_MFEM
+  #error "Quest's SamplingShaper tests on mfem meshes require mfem library."
+#else
+  #include "mfem.hpp"
+#endif
+
+#ifdef AXOM_USE_MPI
+  #include <mpi.h>
+#endif
+
+#include <cmath>
+#include <string>
+#include <iostream>
+
+namespace klee = axom::klee;
+namespace primal = axom::primal;
+namespace quest = axom::quest;
+namespace sidre = axom::sidre;
+namespace slic = axom::slic;
+
+namespace
+{
+const std::string unit_circle_contour =
+  "piece = circle(origin=(0cm, 0cm), radius=1cm, start=0deg, end=360deg)";
+
+constexpr bool very_verbose_output = false;
+
+/// RAII utility class to write a file at construction time and remove it
+/// once the instance is out of scope
+class ScopedTemporaryFile
+{
+public:
+  ScopedTemporaryFile(const std::string& filename, const std::string& contents)
+    : m_filename(filename)
+  {
+    std::ofstream ofs(m_filename.c_str(), std::ios::out);
+    ofs << contents;
+  }
+
+  ~ScopedTemporaryFile()
+  {
+    axom::utilities::filesystem::removeFile(m_filename);
+  }
+
+  const std::string& getFileName() const { return m_filename; }
+
+  std::string getFileContents() const
+  {
+    std::ifstream ifs(m_filename.c_str(), std::ios::in);
+    std::stringstream buffer;
+    buffer << ifs.rdbuf();
+
+    return buffer.str();
+  }
+
+private:
+  const std::string m_filename;
+};
```

Comment:
This little utility class might be more generally useful in support of our internal testing